include/libtorrent/bencode.hpp

```diff
@@ -88,6 +88,28 @@ namespace libtorrent
 
 	namespace detail
 	{
+		template <typename T, typename Cond = typename std::enable_if<
```

Comment:
oh, no.. you just force the template parameter to be a char*.. making this a template seems redundant. couldn't you just make it "inline"?